### PR TITLE
Feature/dag improvements

### PR DESCRIPTION
**Pull request by agduncan94**
The following improvements have been made to the DAG view:
* Reset button
* Start and end nodes for CWL workflows
* Full screen option
* Uses QTip to add tooltips
* Highlight connected edges on hover
* Export as PNG

Look at ICGC-TCGA-PanCancer/OxoG-Dockstore-Tools with workflow path set to /preprocess.vcf for an example.

Also fixes an issue with protractor and Chrome, by now using Firefox.

- [x] Check that you pass the basic compilation and unit tests by running `grunt`

### CODE REVIEW

**Comment by agduncan94** [comment]
I believe the issue we were having with protractor stems from an update to Chrome.

**Comment by agduncan94** [comment]
Found a weird bug where the tooltips persisted between page loads and slowed down the page, but it is now ready for review.

File: app/scripts/controllers/workflowdagview.js [hunk 1]
```diff
@@ -148,13 +148,12 @@ angular.module('dockstore.ui')
...
 
       $scope.clearPopover = function() {
-        $scope.dynamicPopover = {
-          link: '',
-          title: '',
-          type: '',
-          docker: '',
-          run: ''
-        };
+        $scope.dynamicPopover.link = '';
```
Comment (denis-yuen) [inline app/scripts/controllers/workflowdagview.js]: Note to discuss

File: .travis.yml [hunk 2]
```diff
@@ -2,31 +2,36 @@ sudo: true
...
   - 'npm config get registry'
   - 'npm config set registry http://registry.npmjs.org/'
-  - 'export CHROME_BIN=/usr/bin/google-chrome'
   - 'export DISPLAY=:99.0'
   - 'sh -e /etc/init.d/xvfb start'
-  - 'sudo apt-get update'
-  - 'sudo apt-get install -y libappindicator1 fonts-liberation'
-  - 'wget https://dl.google.com/linux/direct/google-chrome-stable_current_amd64.deb'
-  - 'sudo dpkg -i google-chrome*.deb'
+  - 'sleep 3'
```
Comment (denis-yuen) [inline .travis.yml]: Comment on this, why 3?

Comment (agduncan94) [inline .travis.yml]: It was recommended by Travis
https://docs.travis-ci.com/user/gui-and-headless-browsers/#Using-xvfb-to-Run-Tests-That-Require-a-GUI

File: app/templates/registercontainer.html [hunk 3]
```diff
@@ -40,7 +40,7 @@ <h4 class="modal-title">Register Tool</h4>
           ng-if="refreshingContainer">
         <span class="glyphicon glyphicon-refresh glyphicon-refresh-animate">
         </span>
-        Please wait, retrieving Quay.io and/or Docker Hub container...
+        Please wait, retrieving Quay.io and/or Docker Hub tool...
```
Comment (denis-yuen) [inline app/templates/registercontainer.html]: Strictly, I think a tool is either from Quay.io or Docker Hub, but cannot be "and"

File: app/templates/registerworkflow.html [hunk 4]
```diff
@@ -40,7 +40,7 @@ <h4 class="modal-title">Register Workflow</h4>
            ng-if="refreshingWorkflow">
         <span class="glyphicon glyphicon-refresh glyphicon-refresh-animate">
         </span>
-        Please wait, retrieving Bitbucket and/or Github Workflows...
+        Please wait, retrieving Bitbucket, Gitlab and/or Github Workflows...
```
Comment (denis-yuen) [inline app/templates/registerworkflow.html]: Same here, I think workflows are only "or". Or "xor", maybe "one of"

File: app/views/containereditor.html [hunk 5]
```diff
@@ -56,21 +56,21 @@ <h3 class="available-containers">
           ng-if="refreshingContainers">
         <span class="glyphicon glyphicon-refresh glyphicon-refresh-animate">
         </span>
-        Please wait, retrieving Quay.io and/or Docker Hub containers...
+        Please wait, retrieving Quay.io and/or Docker Hub tools...
```
Comment (denis-yuen) [inline app/views/containereditor.html]: "

File: app/views/workfloweditor.html [hunk 6]
```diff
@@ -56,7 +56,7 @@ <h3 class="available-containers">
            ng-if="refreshingWorkflows">
         <span class="glyphicon glyphicon-refresh glyphicon-refresh-animate">
         </span>
-        Please wait, retrieving Github and/or Bitbucket workflows...
+        Please wait, retrieving Github, Gitlab and/or Bitbucket workflows...
```
Comment (denis-yuen) [inline app/views/workfloweditor.html]: "

File: .travis.yml (hunk 2 above)
Comment (denis-yuen) [inline .travis.yml]: Hah! Ok, just link to this

File: app/views/containereditor.html (hunk 5 above)
Comment (agduncan94) [inline app/views/containereditor.html]: This is the general refresh all workflows message, so I think in this case is makes sense to have the and/or.

File: app/views/workfloweditor.html (hunk 6 above)
Comment (agduncan94) [inline app/views/workfloweditor.html]: This is the general refresh all workflows message, so I think in this case is makes sense to have the and/or.

**Pull request merged**